src/cargo/util/toml.rs

```diff
@@ -1040,7 +1051,11 @@ fn normalize(lib: &Option<TomlLibTarget>,
               .set_doctest(toml.doctest.unwrap_or(t2.doctested()))
               .set_benched(toml.bench.unwrap_or(t2.benched()))
               .set_harness(toml.harness.unwrap_or(t2.harness()))
-              .set_for_host(toml.plugin.unwrap_or(t2.for_host()));
+              .set_for_host(match (toml.plugin, toml.rustc_macro) {
+                  (None, None) => t2.for_host(),
+                  (Some(true), _) | (_, Some(true)) => true,
+                  _ => false,
```

Comment:
Oh yes, the latter. Just writing out what the case is catching vs "something we accidentally forgot to cover above"